OneSignalSDK/onesignal/src/main/java/com/onesignal/NotificationRestorer.java

```diff
@@ -183,81 +188,49 @@ private static void skipVisibleNotifications(Context context, StringBuilder dbQu
          return;
 
-      boolean useExtender = (NotificationExtenderService.getIntent(context) != null);
-
+      ArrayList<OneTimeWorkRequest> workRequests = new ArrayList<>();
       do {
-         if (useExtender) {
-            Intent intent = NotificationExtenderService.getIntent(context);
-            addRestoreExtras(intent, cursor);
-            NotificationExtenderService.enqueueWork(context,
-                  intent.getComponent(),
-                  NotificationExtenderService.EXTENDER_SERVICE_JOB_ID,
-                  intent,
-                  false);
-         }
-         else {
-            Intent intent = addRestoreExtras(new Intent(), cursor);
-            ComponentName componentName = new ComponentName(context, RestoreJobService.class);
-            RestoreJobService.enqueueWork(context, componentName, RestoreJobService.RESTORE_SERVICE_JOB_ID, intent, false);
-         }
-
-         if (delay > 0)
-            OSUtils.sleep(delay);
+         int existingId = cursor.getInt(cursor.getColumnIndex(NotificationTable.COLUMN_NAME_ANDROID_NOTIFICATION_ID));
+         String fullData = cursor.getString(cursor.getColumnIndex(NotificationTable.COLUMN_NAME_FULL_DATA));
+         long dateTime = cursor.getLong(cursor.getColumnIndex(NotificationTable.COLUMN_NAME_CREATED_TIME));
+
+         Data workData = new Data.Builder()
+                 .putInt(ANDROID_NOTIF_ID_RESTORE_INTENT_KEY, existingId)
+                 .putString(JSON_PAYLOAD_RESTORE_INTENT_KEY, fullData)
+                 .putBoolean(IS_RESTORING_RESTORE_INTENT_KEY, true)
+                 .putLong(TIMESTAMP_RESTORE_INTENT_KEY, dateTime)
+                 .build();
+
+         OneTimeWorkRequest workRequest = new OneTimeWorkRequest.Builder(OSNotificationRestoreManager.NotificationRestoreWorker.class)
+                 .setInputData(workData)
+                 .build();
+
+         workRequests.add(workRequest);
       } while (cursor.moveToNext());
+
+      WorkManager.getInstance(context)
+              .enqueue(workRequests);
    }
 
-   private static Intent addRestoreExtras(Intent intent, Cursor cursor) {
-      int existingId = cursor.getInt(cursor.getColumnIndex(NotificationTable.COLUMN_NAME_ANDROID_NOTIFICATION_ID));
-      String fullData = cursor.getString(cursor.getColumnIndex(NotificationTable.COLUMN_NAME_FULL_DATA));
-      Long datetime = cursor.getLong(cursor.getColumnIndex(NotificationTable.COLUMN_NAME_CREATED_TIME));
+   private static final int RESTORE_NOTIFICATIONS_DELAY = 15; // seconds
+   static void startDelayedRestoreTaskFromWorkManager(Context context) {
+      OneSignal.Log(OneSignal.LOG_LEVEL.INFO, "");
 
-      intent.putExtra("json_payload", fullData)
-            .putExtra("android_notif_id", existingId)
-            .putExtra("restoring", true)
-            .putExtra("timestamp", datetime);
+      Constraints workConstraints = new Constraints.Builder()
+              .setTriggerContentMaxDelay(RESTORE_NOTIFICATIONS_DELAY, TimeUnit.SECONDS)
```

Comment:
This should be a min instead of a max. We want to delay the job from running as on app upgrade some resources such as icon won't be available right away.